src/containers/category-list/CategoryList.jsx

```diff
@@ -0,0 +1,103 @@
+import { Box } from '@mui/material'
+import { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSearchParams } from 'react-router-dom'
+import AppButton from '~/components/app-button/AppButton'
+import CategoryCard from '~/components/category-card/CategoryCard'
+import Loader from '~/components/loader/Loader'
+import { styles } from '~/containers/category-list/CategoryList.styles'
+import useAxios from '~/hooks/use-axios'
+import { categoryService } from '~/services/category-service'
+
+// TODO: update path of category images in db and import images
+import categoryImg from '~/assets/img/student-home-page/service_icon.png'
+
+const itemsPerPage = 24
+
+const CategoryList = () => {
+  const { t } = useTranslation()
+  const [searchParams] = useSearchParams()
+  const [categories, setCategories] = useState([])
+  const [page, setPage] = useState(1)
```

Comment:
What do you think about keeping the page number in the URL? Or have you discussed this implementation?

Reply:
I haven't discussed the implementation of pages for this component, but I think in this case it's better to keep the page value in the state. I use it only for calculating how many items need to be skipped when the user clicks the "load more" button. In case when I keep it in the URL, the user can edit it and I don't think it's convenient for this type of pagination. But I am open to new ideas